### PR TITLE
[ci:component:github.com/gardener/etcd-druid:v0.1.10->v0.1.11]

### PR DESCRIPTION
**Pull request by gardener-robot-ci-2**
*Release Notes*:
``` improvement operator github.com/gardener/etcd-druid #34 @georgekuruvillak
Changed logic in etcd druid to query for statefulset readiness and update the etcd status accordingly
```

### CODE REVIEW

File: pkg/operation/botanist/waiter.go [hunk 1]
```diff
@@ -84,7 +84,7 @@ func (b *Botanist) WaitUntilEtcdReady(ctx context.Context) error {
 				lastErrors = multierror.Append(lastErrors, fmt.Errorf("%s reconciliation in process", etcd.Name))
 			case etcd.Status.LastError != nil:
 				lastErrors = multierror.Append(lastErrors, fmt.Errorf("%s reconciliation errored with %s", etcd.Name, err))
-			case !etcd.Status.Ready:
+			case etcd.Status.Ready == nil || !*etcd.Status.Ready:
```
Comment (timuthy) [inline pkg/operation/botanist/waiter.go]: Nit

```suggestion
			case !utils.IsTrue(etcd.Status.Ready):
```

with `"github.com/gardener/gardener/pkg/utils"`

File: pkg/utils/kubernetes/health/health.go [hunk 2]
```diff
@@ -143,7 +143,7 @@ func CheckStatefulSet(statefulSet *appsv1.StatefulSet) error {
 // CheckEtcd checks whether the given Etcd is healthy.
 // A Etcd is considered healthy if its ready field in status is true.
 func CheckEtcd(etcd *druidv1alpha1.Etcd) error {
-	if !etcd.Status.Ready {
+	if etcd.Status.Ready == nil || !*etcd.Status.Ready {
```
Comment (timuthy) [inline pkg/utils/kubernetes/health/health.go]: Nit

```suggestion
			case !utils.IsTrue(etcd.Status.Ready):
```

with `"github.com/gardener/gardener/pkg/utils"`

**Review by timuthy — COMMENTED** [review]
Just two nits. I leave it up to you if you want to address them. LGTM otherwise.

**Review by swapnilgm — APPROVED** [review]
lgtm

**Review by georgekuruvillak — APPROVED** [review]

**Comment by georgekuruvillak** [comment]
lgtm

**Pull request merged**